plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions.php

```diff
@@ -78,11 +78,27 @@ protected function render( $attributes, $content, $block ) {
 
 		$product_type = $product->get_type();
 
+		$slug = $product_type . '-product-add-to-cart-with-options';
+
 		if ( in_array( $product_type, array( ProductType::SIMPLE, ProductType::EXTERNAL, ProductType::VARIABLE, ProductType::GROUPED ), true ) ) {
+			$template_part_path = Package::get_path() . 'templates/' . BlockTemplateUtils::DIRECTORY_NAMES['TEMPLATE_PARTS'] . '/' . $slug . '.html';
+		} else {
+			/**
+			 * Filter to declare product type's cart block template is supported.
+			 * 
+			 * @since x.x.x
+			 * @param boolean $supports Is the type supported
+			 * @param string $product_type The product type 
+			 */
+			$template_part_path = apply_filters( 'woocommerce_' . $product_type . '_add_to_cart_with_options_block_template_part', Package::get_path() . 'templates/' . BlockTemplateUtils::DIRECTORY_NAMES['TEMPLATE_PARTS'] . '/' . $slug . '.html', $product_type );
```

Comment:
We know the default path (`Package::get_path() . 'templates/' . BlockTemplateUtils::DIRECTORY_NAMES['TEMPLATE_PARTS'] . '/' . $slug . '.html'`) will never return a file because WC core won't include templates for 3rd-party product types. So I think we could simply do:

```suggestion
			$template_part_path = apply_filters( '__experimental_woocommerce_' . $product_type . '_add_to_cart_with_options_block_template_part', false, $product_type );
```

I would also like to prefix the hook as `__experimental` for now. At least until we get it working in the editor side. :slightly_smiling_face: